pkg/process/procutil/process_windows.go

```diff
@@ -497,14 +497,29 @@ func fillProcessDetails(pid int32, proc *Process) error {
 	}
 	proc.Username = userName
 
-	cmdParams := getProcessCommandParams(procHandle)
+	imagePath, err := winutil.GetImagePathForProcess(procHandle)
+	if err != nil {
+		log.Debugf("Error retrieving exe path for pid %v %v", pid, err)
+	}
+	proc.Exe = imagePath
+
+	// we cannot read the command line if the process is protected
+	var cmdLine string
+	if !isProtected {
+		processCmdParams, err := winutil.GetCommandParamsForProcess(procHandle, false)
+		if err != nil {
+			log.Debugf("Error retrieving full command line %v", err)
+		}
 
-	proc.Cmdline = ParseCmdLineArgs(cmdParams.CmdLine)
-	if len(cmdParams.CmdLine) > 0 && len(proc.Cmdline) == 0 {
-		log.Warnf("Failed to parse the cmdline:%s for pid:%d", cmdParams.CmdLine, pid)
+		if processCmdParams != nil {
+			cmdLine = processCmdParams.CmdLine
+		}
 	}
 
-	proc.Exe = cmdParams.ImagePath
+	proc.Cmdline = ParseCmdLineArgs(cmdLine)
```

Comment:
Suggestion: since cmdLine is only ever populated if `!isProtected`, we can move this assignment and the following check into the if block above.